sockeye/model.py

```diff
@@ -189,6 +191,19 @@ def __init__(self,
 
         self.knn : Optional[layers.KNN] = None
 
+    def set_inference_only(self, inference_only: bool):
+        """
+        Turn inference_only/eval mode on or off.
+        """
+        if self.inference_only != inference_only:
```

Comment:
Is this check needed? Does setting `inference_only` to the same value multiple times have any negative effects?